### PR TITLE
Ajc experiment docs

### PR DESCRIPTION
**Pull request by ambrosejcarr**
- Links Experiment and Field of View to the docs. 
- Refactors some docstrings in the above files so sphinx doesn't choke on them (with some textual changes, please check my modifications). 

Also I discovered how to make internal links with ```:ref:`ReferenceName` ```. I'll start weaving those into the docs.

### CODE REVIEW

File: docs/source/API/core_objects/index.rst [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
+:ref:`FieldOfView` objects, and a :ref:`Codebook`, which maps detected spots to the entities they
+target.
+
+Each :ref:`FieldOfView` consists of a set of Primary Images and optionally, Auxiliary images that
+may contain information on nuclei (often used to seed segmentation) or fiduciary beads (often used
+to enable fine registration).
+
+Both Primary and Auxiliary Images are referenced by slicedimage_ TileSet_ objects, which map
+two dimensional image tiles stored on disk into a 5-dimensional Image Tensor that labels each
+``(z, y, z)`` tile with the ``round`` and ``channel`` that it corresponds to. When loaded into
```
Comment (shanaxel42) [inline docs/source/API/core_objects/index.rst]: (x,y,z)?

Comment (ambrosejcarr) [reply docs/source/API/core_objects/index.rst]: (z, y, x)! thanks.

File: docs/source/API/core_objects/index.rst [hunk 2]
```diff
@@ -1,12 +1,48 @@
-.. _coreobjects
-
 Core Objects
 ============
 
+The top-level object in a starfish workflow is the :ref:`Experiment`. It is composed of one or more
+:ref:`FieldOfView` objects, and a :ref:`Codebook`, which maps detected spots to the entities they
```
Comment (ttung) [inline docs/source/API/core_objects/index.rst]: s/spots/features/

File: docs/source/API/core_objects/index.rst [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+target.
+
+Each :ref:`FieldOfView` consists of a set of Primary Images and optionally, Auxiliary images that
```
Comment (ttung) [inline docs/source/API/core_objects/index.rst]: inconsistent casing (Images vs images)

File: docs/source/API/core_objects/index.rst [hunk 4]
```diff
@@ -1,12 +1,48 @@
...
+may contain information on nuclei (often used to seed segmentation) or fiduciary beads (often used
```
Comment (ttung) [inline docs/source/API/core_objects/index.rst]: are these the only types of aux images?  Would clarify the wording to indicate more possibilities.

File: docs/source/API/core_objects/index.rst [hunk 5]
```diff
@@ -1,12 +1,48 @@
...
+to enable fine registration).
+
+Both Primary and Auxiliary Images are referenced by slicedimage_ TileSet_ objects, which map
+two dimensional image tiles stored on disk into a 5-dimensional Image Tensor that labels each
+``(z, y, x)`` tile with the ``round`` and ``channel`` that it corresponds to. When loaded into
```
Comment (ttung) [inline docs/source/API/core_objects/index.rst]: tiles are only 2d.

File: docs/source/API/core_objects/index.rst [hunk 6]
```diff
@@ -1,12 +1,48 @@
...
+memory, these Image Tensors are stored in :ref:`ImageStack` objects. The :ref:`ImageStack` is what
+starfish uses to execute image pre-processing, and serves as the substrate for spot finding.
```
Comment (ttung) [inline docs/source/API/core_objects/index.rst]: spot or feature?

File: starfish/experiment/experiment.py [hunk 7]
```diff
@@ -20,12 +20,16 @@ class FieldOfView:
...
-    Properties
-    -------
-    name                   The name of the FOV.  In an experiment with only a single FOV, this may
-                           be None.
-    primary_image          The primary image for this field of view.
-    auxiliary_image_types  A set of all the auxiliary image types.
+    Attributes
+    ----------
+    name : str
+        The name of the FOV.  In an experiment with only a single FOV, this may be None.
```
Comment (ttung) [inline starfish/experiment/experiment.py]: Second sentence should be removed.

File: starfish/experiment/experiment.py [hunk 8]
```diff
@@ -101,24 +105,27 @@ class Experiment:
...
-                  FOVs, use this cautiously.
-    fovs          Given a callable that accepts a FOV, return all the FOVs that the callable returns
-                  True when passed the FOV.
-    fovs_by_name  Given one or more FOV names, return the FOVs that match those names.
-
-    Properties
-    -------
-    codebook      Returns the codebook associated with this experiment.
-    extras        Returns the extras dictionary associated with this experiment.
+    from_json()
```
Comment (ttung) [inline starfish/experiment/experiment.py]: is there a way to distinguish this as a classmethod?

File: docs/source/API/core_objects/index.rst (hunk 2 above)
Comment (ambrosejcarr) [reply docs/source/API/core_objects/index.rst]: I don't think we actually generalize beyond spots at the moment. @dganguli ?

File: starfish/experiment/experiment.py (hunk 7 above)
Comment (ttung) [reply starfish/experiment/experiment.py]: This is corrected in #701, so don't bother fixing it. :)

Comment (ambrosejcarr) [reply starfish/experiment/experiment.py]: To be clear: this comment doesn't need to be addressed?

**Pull request merged**